### PR TITLE
Refinement of Spinel Protocol Documentation

### PR DESCRIPTION
**Pull request by darconeous**
This commit refines the documentation of the Spinel NCP protocol. What was previously a single markdown-based document has now been broken into multiple files which can be edited independently. These files can then be "compiled" into a final document in either XML, HTML, or
plaintext formats. The [`mmark`](https://github.com/miekg/mmark) is used for compiling the source documentation into a single XML file and [`xml2rfc`](https://xml2rfc.tools.ietf.org/) is used to convert from the XML to the final desired format.

Note that the building of the protocol documents is handled independently from the normal OpenThread build. The spinel protocol document source files are intentionally excluded from the build system, since there is no value in including them. Copies of the final compiled documents are included to ensure that they are distributed with any release archives.

This commit is focused on introducing the new structure the protocol document. The document itself is a work in progress and may contain errors or require revisions, since the `draft-` prefix in the filenames.

This commit is related to issue #492, "☂ Spinel NCP Protocol Standardization".

### CODE REVIEW

**Comment by darconeous** [comment]
For your convenience, you can preview the HTML version of the document [here](https://rawgit.com/darconeous/openthread/811336f5d171a79a976bd394cfc984fcbd845962/doc/draft-spinel-protocol.html).

**Comment by nibanks** [comment]
HTML version is cool and easy to read! I have been using it to better my understanding of the Spinel protocol. 👍

**Comment by darconeous** [comment]
@jwhui, thoughts on merging?

**Comment by abtink** [comment]
LGTM :+1:

File: src/ncp/PROTOCOL.md [hunk 1]
```diff
@@ -1,1712 +1,7 @@
...
-* Type: Read-Write
-* Packed-Encoding: `b`
-
-Allow the NCP to directly respond to ICMP ping requests. If this is
-turned on, ping request ICMP packets will not be passed to the host.
-
-Default value is `false`.
+This document has moved into the top level `doc` folder and renamed
+[`draft-spinel-protocol.txt`](../../doc/draft-spinel-protocol.txt)
+and [`draft-spinel-protocol.html`](../../doc/draft-spinel-protocol.txt).
```
Comment (jwhui) [inline src/ncp/PROTOCOL.md]: Did you intend to link the `draft-spinel-protocol.html` here?

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+* Type: Read-Only
+* Packed-Encoding: `D`
+
+### PROP 89: PROP_THREAD_STABLE_NETWORK_DATA_VERSION
+* Type: Read-Only
+* Packed-Encoding: `S`
+
+### PROP 90: PROP_THREAD_ON_MESH_NETS
+* Type: Read-Write
+* Packed-Encoding: `A(T(6CbC))`
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: FYI, #537 references this line.

**Comment by jwhui** [comment]
This looks great overall.  See my minor comments above.

File: src/ncp/PROTOCOL.md (hunk 1 above)
Comment (darconeous) [inline src/ncp/PROTOCOL.md]: Oops. Will fix.

**Comment by darconeous** [comment]
@jwhui: Addressed those issues.

File: doc/spinel-protocol-src/spinel-test-vectors.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    84 02 5A
+
+## Test Vector: Returned list of on-mesh networks
+
+* IID: 0
+* TID: 4
+* CMD: 6 (`CMD_VALUE_IS`)
+* PROP: 90 (`PROP_THREAD_ON_MESH_NETS`)
+* VALUE: Array of structures, encoded as `A(T(6CbC))`
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: Should this be `A(T(6CbCb))`?

File: doc/spinel-protocol-src/spinel-test-vectors.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+<!-- TODO: This test vector is incomplete. -->
+
+## Test Vector: Adding an on-mesh network
+
+* IID: 0
+* TID: 5
+* CMD: 4 (`CMD_VALUE_INSERT`)
+* PROP: 90 (`PROP_THREAD_ON_MESH_NETS`)
+* VALUE: Structure, encoded as `6CbC`
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: Should this be `A(T(6CbCb))`?

File: doc/spinel-protocol-src/spinel-test-vectors.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+<!-- RQ -- FIXME: This test vector is incomplete. -->
+
+## Test Vector: Insertion notification of an on-mesh network
+
+* IID: 0
+* TID: 5
+* CMD: 7 (`CMD_VALUE_INSERTED`)
+* PROP: 90 (`PROP_THREAD_ON_MESH_NETS`)
+* VALUE: Structure, encoded as `6CbC`
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: Should this be `A(T(6CbCb))`?

**Comment by mszczodrak** [comment]
LGTM :+1:

File: doc/spinel-protocol-src/spinel-test-vectors.md (hunk 5 above)
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: Let's fix this in a separate commit. The documentation had many typos to begin with and this commit is about restructuring the document, not fixing all errors.

File: doc/spinel-protocol-src/spinel-test-vectors.md (hunk 4 above)
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: See https://github.com/openthread/openthread/pull/534#discussion_r78218575

File: doc/spinel-protocol-src/spinel-test-vectors.md (hunk 3 above)
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-test-vectors.md]: See https://github.com/openthread/openthread/pull/534#discussion_r78218575

**Comment by darconeous** [comment]
See https://github.com/openthread/openthread/pull/534#discussion_r78218575

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**